docs/getting-started.md

```diff
@@ -511,42 +511,101 @@ const messageForClaimer = new Kilt.Message(
-The result will be a boolean indicating the result of the verification and the attested claim(s) which are either sent in their entirety OR have been stripped off of the properties that the verifier did not request to verify.
+When verifying the claimer's message, the verifier checks that 3 important requirements are met:
+
+- The credential is valid, i.e. it has been registered to the public ledger, has not been revoked, and the data has not been tampered with.
+- The attester who registered the credential is one that this verifier trusts.
+- The submission is coming from the legitimate owner of this credential, i.e. the credential has not been stolen.
 
 ```typescript
+const messageForVerifier = Kilt.Message.decrypt(encrypted, verifier)
+// Our messages are signed, so we authenticate the sender already during decryption, which internally calls:
+Kilt.Message.ensureHashAndSignature(encrypted, claimer.address)
+Kilt.Message.ensureOwnerIsSender(messageForVerifier)
+
 if (
   messageForVerifier.body.type ===
   Kilt.Message.BodyType.SUBMIT_CLAIMS_FOR_CTYPES
 ) {
   const claims = messageForVerifier.body.content
-  const isValid = await Kilt.AttestedClaim.fromAttestedClaim(claims[0]).verify()
-  console.log('Verifcation success?', isValid)
   console.log('Attested claims from verifier perspective:\n', claims)
+
+  const credential = Kilt.AttestedClaim.fromAttestedClaim(claims[0])
+
+  // credential is registered on chain, has not been tampered with, and has not been revoked
+  const isValid = await credential.verify()
+  console.log('Verification success?', isValid)
+
+  // AttestedClaim exposes attester address to compare against a whitelist of trusted attesters
+  console.log('Attester address:', credential)
 }
 ```
 
+### 6.3. Replay Protection
+
+In certain use cases, an attacker may intercept and copy credential submissions traveling from claimers to verifiers in an attempt to convince the verifier to accept the credential submission again later on.
+To give an example for illustration purposes only, think of a turnstile that allows passage only upon presentation of a valid access credential.
+To prevent these types of attacks, KILT messages are timestamped and expose a unique identifier (the message hash).
+Verifiers should impose limits on the acceptable range for these timestamps and keep a record of previous submissions, which can be purged after their acceptance range has run out.
+
+Define acceptance range and set up a record of past submissions:
+
+```typescript
+const MAX_ACCEPTED_AGE = 60_000 // ms -> 1 minute
+const MIN_ACCEPTED_AGE = -1_000 // allow for some imprecision in system time
+const submissions = new Map<string, number>()
+```
+
+Check record for each incoming message and update if accepted:
+
+```typescript
+// is hash fresh and createdAt recent ?
+if (
+  submissions.has(encrypted.hash) ||
+  encrypted.createdAt > Date.now() + MAX_ACCEPTED_AGE ||
```

Comment:
Does this not have to be
```suggestion
  encrypted.createdAt < Date.now() - MAX_ACCEPTED_AGE ||
```
?
Since I want to accept messages created up to one minute in the past

Reply:
oh, true